addons/stock_landed_costs/tests/test_stock_landed_costs_purchase.py

```diff
@@ -0,0 +1,246 @@
+        #         -------------------------------------------
+        #         1.labour            -5        By Equal
+        #         2.brokerage         -50       By Quantity
+        #         3.transportation    -50       By Weight
+        #         4.packaging         -5        By Volume
+
+        # Process incoming shipment
+        income_ship = self._process_incoming_shipment()
+        # Refrigerator outgoing shipment.
+        self._process_outgoign_shipment()
+        # Apply landed cost for incoming shipment.
+        stock_landed_cost = self._create_landed_costs({
+            'equal_price_unit': 10,
+            'quantity_price_unit': 150,
+            'weight_price_unit': 250,
+            'volume_price_unit': 20}, income_ship)
+        # Compute landed costs
+        stock_landed_cost.compute_landed_cost()
+        valid_vals = {
+            'equal': 5.0,
+            'by_quantity_refrigerator': 50.0,
+            'by_quantity_oven': 100.0,
+            'by_weight_refrigerator': 50.0,
+            'by_weight_oven': 200.0,
+            'by_volume_refrigerator': 5.0,
+            'by_volume_oven': 15.0}
+        # Check valuation adjustment line recognized or not
+        self._validate_additional_landed_cost_lines(stock_landed_cost, valid_vals)
+        # Validate the landed cost.
+        stock_landed_cost.button_validate()
+        self.assertTrue(stock_landed_cost.account_move_id, 'Landed costs should be available account move lines')
+        # Create negative landed cost for previously incoming shipment.
+        stock_negative_landed_cost = self._create_landed_costs({
+            'equal_price_unit': -5,
+            'quantity_price_unit': -50,
+            'weight_price_unit': -50,
+            'volume_price_unit': -5}, income_ship)
+        # Compute negative landed costs
+        stock_negative_landed_cost.compute_landed_cost()
+        valid_vals = {
+            'equal': -2.5,
+            'by_quantity_refrigerator': -16.67,
+            'by_quantity_oven': -33.33,
+            'by_weight_refrigerator': -10.00,
+            'by_weight_oven': -40.00,
+            'by_volume_refrigerator': -1.25,
+            'by_volume_oven': -3.75}
+        # Check valuation adjustment line recognized or not
+        self._validate_additional_landed_cost_lines(stock_negative_landed_cost, valid_vals)
+        # Validate the landed cost.
+        stock_negative_landed_cost.button_validate()
+        self.assertEqual(stock_negative_landed_cost.state, 'done', 'Negative landed costs should be in done state')
+        self.assertTrue(stock_negative_landed_cost.account_move_id, 'Landed costs should be available account move lines')
+        account_entry = self.env['account.move.line'].read_group(
+            [('move_id', '=', stock_negative_landed_cost.account_move_id.id)], ['debit', 'credit', 'move_id'], ['move_id'])[0]
+        self.assertEqual(account_entry['debit'], account_entry['credit'], 'Wrong Account Entry')
```

Comment:
Should check amounts on account_move_lines here too. (normally also the accounts used)